src/internal/util/toSubscriber.ts

```diff
@@ -24,3 +25,29 @@ export function toSubscriber<T>(
 
   return new Subscriber(nextOrObserver, error, complete);
 }
+
+function isObserver<T>(value: any): value is Observer<T> {
+  return value && typeof value.next === 'function' && typeof value.error === 'function' && typeof value.complete === 'function';
+}
+
+function isSubscriber<T>(value: any): value is Subscriber<T> {
+  return value instanceof Subscriber || (isObserver(value) && isSubscription(value));
+}
+
+/**
+ * Used to wrap objects that we know to be full observers
+ * having a `next`, `error`, and `complete` method.
+ */
+class FullObserverSubscriber<T> extends Subscriber<T> {
+  /**
+   * Creates an instance of a subscriber
+   *
+   * NOTE: The `protected destination` is a tricky override of the
+   * destination on the original `Subscriber` class. This whole
+   * class may look unnecessary, but it's doing this trick to get around
+   * some of the ugly inheritance stuff we have going on in the library.
+   */
```

Comment:
So the purpose of `destination`, here, is to ensure the instance's `destination` property is the full observer that was passed to the ctor, bypassing the `SafeSubscriber` wrapping that would happen within the `Subscriber` implementation?

Reply:
That's right. There was a weird round-about way of doing this before through the `SubjectSubscriber`.